test/fixture/locators.test.ts

```diff
@@ -173,5 +173,35 @@ test.describe('lib/fixture.ts (locators)', () => {
     })
   })
 
-  // TDOO: deferred page (do we need some alternative to `findBy*`?)
+  test.describe('deferred page', () => {
+    test.beforeEach(async ({page}) => {
+      await page.goto(`file://${path.join(__dirname, '../fixtures/late-page.html')}`)
+    })
+
+    test.afterEach(async ({page}) => page.close())
+
+    test('should handle the findBy* methods', async ({queries}) => {
+      const locator = await queries.findByText('Loaded!', undefined, {timeout: 7000})
+
+      expect(await locator.textContent()).toEqual('Loaded!')
+    })
+
+    test('should handle the findAllBy* methods', async ({queries}) => {
+      const locator = await queries.findAllByText(/Hello/, undefined, {timeout: 7000})
+
+      const text = await Promise.all([locator.nth(0).textContent(), locator.nth(1).textContent()])
+
+      expect(text).toEqual(['Hello h1', 'Hello h2'])
+    })
+
```

Comment:
If we're going to support [`Locator#waitFor.state`](https://playwright.dev/docs/api/class-locator#locator-wait-for) we should add an extra `<div>` element with a style of `visibility: hidden` into `fixtures/late-page.html` which is not added within a timeout and is instead pre-existing within `<body>`. That way we could test `state: 'hidden'`.

Reply:
Does it make any sense to support hidden and detached just because waitFor supports them? [playwright.dev/docs/api/class-locator#locator-wait-for](https://playwright.dev/docs/api/class-locator#locator-wait-for)
https://github.com/testing-library/playwright-testing-library/pull/488/files#diff-04543a94352d1484d0a0add940de7e7b20b0e8d7d58ca1e3aa8d705ba7de95cdR55-R58

Reply:
I went ahead and removed `hidden` and `detached` from the types for now. Unfortunately our follow-up `getBy` query for the Testing Library error kinda breaks down when the element is attached but not visible because the `getBy` query doesn't throw an error in that case 😑.